lib/spack/spack/cmd/__init__.py

```diff
@@ -109,15 +106,19 @@ def parse_specs(args, **kwargs):
             elif normalize:
                 spec.normalize()
 
-        return specs
-
     except spack.parse.ParseError as e:
-        tty.error(e.message, e.string, e.pos * " " + "^")
-        sys.exit(1)
+        tty.die(e.message, e.string, e.pos * " " + "^")
 
     except spack.spec.SpecError as e:
-        tty.error(e.message)
-        sys.exit(1)
+        tty.die(e.message)
+
+    if allow_multi:
+        return specs
+
+    if len(specs) != 1:
+        tty.die('only one spec can be installed at a time.')
+    spec = specs.pop()
```

Comment:
`allow_multi` should be irrelevant now, the parser accepts multiple specs as of #2769